src/stdlib_stringlist_type.f90

```diff
@@ -710,26 +736,182 @@ end subroutine insert_before_stringarray_int_impl
+
+        type(string_type), allocatable                          :: get_range_idx_impl(:)
+
+        call get_engine( list, first, last, get_range_idx_impl )
+
+    end function get_range_idx_impl
+
+  ! pop & drop:
+
+    !> Version: experimental
+    !>
+    !> Removes strings present at indexes in interval ['first', 'last']
+    !> Stores captured popped strings in array 'capture_popped'
+    !> No return
+    pure subroutine pop_drop_engine( list, first, last, capture_popped )
+        class(stringlist_type), intent(inout)                   :: list
+        type(stringlist_index_type), intent(in)                 :: first, last
+        type(string_type), allocatable, intent(out), optional   :: capture_popped(:)
+
+        integer                                                 :: firstn, lastn, from, to
+        integer                                                 :: i, inew, pos, old_len, new_len
+        type(string_type), dimension(:), allocatable            :: new_stringarray
+
+        old_len = list%len()
+        firstn  = list%to_current_idxn( first )
+        lastn   = list%to_current_idxn( last )
+        from    = max( firstn, 1 )
+        to      = min( lastn, old_len )
+        
+        ! out of bounds indexes won't modify stringlist
+        if ( from <= to ) then
+            pos     = to - from + 1
+            new_len = old_len - pos
 
-        integer                                 :: idxn
+            allocate( new_stringarray(new_len) )
+            do i = 1, from - 1
+                call move( list%stringarray(i), new_stringarray(i) )
+            end do
 
-        idxn = list%to_current_idxn( idx )
+            ! capture popped strings
+            if ( present(capture_popped) ) then
+                call get_engine( list, shift( first, from - firstn ), &
+                    & shift( last, lastn - to ), capture_popped )
+            end if
 
-        ! if the index is out of bounds, return a string_type equivalent to empty string
-        if ( 1 <= idxn .and. idxn <= list%len() ) then
-            get_string_idx_wrap = list%stringarray(idxn)
+            inew = from
+            do i = to + 1, old_len
+                call move( list%stringarray(i), new_stringarray(inew) )
+                inew = inew + 1
+            end do
 
+            call move_alloc( new_stringarray, list%stringarray )
+        else
+            if ( present(capture_popped) ) then
+                allocate( capture_popped(0) )
+            end if
         end if
 
-    end function get_string_idx_wrap
+    end subroutine pop_drop_engine
+
+    !> Version: experimental
+    !>
+    !> Removes the string present at stringlist_index 'idx' in stringlist 'list'
+    !> Returns the removed string
+    function pop_idx_impl( list, idx )
+        class(stringlist_type), intent(inout)           :: list
+        type(stringlist_index_type), intent(in)         :: idx
+        type(string_type)                               :: pop_idx_impl
```

Comment:
What `_impl` means here (and in other places)? It is a bit obscure to me.

Reply:
`impl` means implementation.

Reply:
I believe it's redundant and can be cause of potential confusion. I would suggest to remove the suffix.

Reply:
I will have to somehow differentiate ti from `pop_idx`.

Reply:
I think the short form of the type bound procedure is not necessary in this case: I would only use `pop_idx` or `pop_index`. The same goes with all the other TBPs with the `_impl` suffix.

Reply:
I didn't understand it @epagone, can you please elaborate on this? May be you can **insert a suggestion** and then I will take it forward.

Reply:
Ok, sure :+1: